controllers/purchaseorder.js

```diff
@@ -0,0 +1,224 @@
+						query+= " then";
+						query+= " `user`.`user_name`";
+						query+= " else";
+						query+= " `group`.`group_name` end";
+						query+= " as `assigned_to`";
+						query+= " from `purchase_order`";
+						query+= " inner join `purchase_order_address` on `purchase_order_address`.`idpurchase_order` = `purchase_order`.`idpurchase_order`";
+						query+= " inner join `purchase_order_custom_fld` on `purchase_order_custom_fld`.`idpurchase_order` = `purchase_order`.`idpurchase_order`";
+						query+= " left join `user` on `user`.`iduser` = `purchase_order`.`iduser`";
+						query+= " left join `purchase_order_to_grp_rel` on `purchase_order_to_grp_rel`.`idpurchase_order` = `purchase_order`.`idpurchase_order`";
+						query+= " left join `vendor` on `vendor`.`idvendor` = `purchase_order`.`idvendor`";
+						query+= " left join `group` on `group`.`idgroup` = `purchase_order_to_grp_rel`.`idgroup`";
+						query+= " left join `contacts` on `contacts`.`idcontacts` = `purchase_order`.`idcontacts`";
+						query+= " where `purchase_order`.`deleted` = 0";
+						query+= whereClause;
+						query+= " order by `purchase_order`.`idpurchase_order`";
+						query+= " limit "+page+" , "+limit;
+					
+					app.models.purchaseorder
+					.query(query, function(err, purchaseorder) {
+						if (err) return autoCallback(err);
+						   
+						return autoCallback(null, purchaseorder);
+					});
+				}]
+			},
+			function(err, results) {
+				if (err) return next(err);
+				var purchaseorders = {},
+					purchseorderPrefix = results.getPurchaseOrderPrifix;
+				
+				async.forEachOf(results.getRecords, function(purchaseorder, key, eachCallBack) {
+					var addressInfo = {};
+					
+					async.auto({
+						// set the quote address
+						setAddress: function(modifiedCallBack) {
+							modulesConfig.moduleAttributes.PurchaseOrder.address.forEach( function(attr) {
+								addressInfo[attr]= purchaseorder[attr];
+							});
+							purchaseorder.address = addressInfo;
+							return modifiedCallBack();
+						},
+						// set the tax and quote number
+						setTaxAndPurchaseOrderNumValues: function(modifiedCallBack) {
+							_.map(purchaseorder, function(v, k) {
+								// making sure that the tax_values and shipping_handling_tax_values are parsed into proper format
+								if (k === 'tax_values' || k === 'shipping_handling_tax_values') {
+									v = common.parseTaxData(v);
+									purchaseorder[k] = v;
+								}
+								// prefixing the quote number with the quote prefix from setting
+								if (purchaseorderPrefix && purchaseorderPrefix.settingData && k === 'po_number') {
+									v= purchaseorderPrefix.settingData+''+v;
+									purchaseorder[k] = v;
+								}
+							});
+							return modifiedCallBack();
+						},
+						// set the line items which is one-to-many relation with quote
+						setLineItems: function(modifiedCallBack) {
+							app.models.lineitems 
+							.find({recordid:purchase_order.id,moduleId:16})
```

Comment:
`urchase_order.id` is undefined, and the loop is going as 
```
async.forEachOf(results.getRecords, function(purchaseorder, key, eachCallBack) {
```
So it should be as

```
.find({recordid:purchaseorder.id,moduleId:16})
```
Also all the variables, functions etc should be `camelCase` not `snake case`